google/oauth2/credentials.py

```diff
@@ -120,3 +126,55 @@ def refresh(self, request):
         self.expiry = expiry
         self._refresh_token = refresh_token
         self._id_token = grant_response.get('id_token')
+
+    @classmethod
+    def from_authorized_user_info(cls, info, scopes=None):
+        """Creates a Credentials instance from parsed authorized user info.
+
+        Args:
+            info (Mapping[str, str]): The authorized user info in Google
+                format.
+            scopes (Sequence[str]): Optional list of scopes to include in the
+                credentials.
+
+        Returns:
+            google.oauth2.credentials.Credentials: The constructed
+                credentials.
+
+        Raises:
+            ValueError: If the info is not in the expected format.
+        """
+        try:
```

Comment:
We already have all of this logic [here](https://github.com/GoogleCloudPlatform/google-auth-library-python/blob/master/google/auth/_cloud_sdk.py#L85). Can you think of de-duplicating it?

Reply:
Done